lib/instrumentation/modules/http2.js

```diff
@@ -68,6 +78,61 @@ module.exports = function (http2, agent, { enabled }) {
         eos(stream, function () {
           trans.end()
         })
+      } else if (event === 'request' && reqIsHTTP1(stream)) {
+        // http2.createSecureServer() supports a `allowHTTP1: true` option.
+        // When true, an incoming client request that supports HTTP/1.x but not
+        // HTTP/2 will be allowed. It will result in a 'request' event being
+        // emitted. We wrap that here.
+        //
+        // Note that, a HTTP/2 request results in a 'stream' event (wrapped
+        // above) *and* a 'request' event. We do not want to wrap the
+        // compatibility 'request' event in this case. Hence the `reqIsHTTP1`
+        // guard.
+        const req = stream
+        const res = headers
+
+        agent.logger.debug('intercepted request event call to http2.Server.prototype.emit for %s', req.url)
+        // TODO: handle ignoring requests as per isRequestBlacklisted in http-shared.js
```

Comment:
REVIEW NOTE: Alan, are you okay with this work being deferred to a separate ticket? AFAICT the regular http2 code path above *already* does not handle ignoring some requests per isRequestBlacklisted.

Reply:
> Alan, are you okay with this work being deferred to a separate ticket?

Yup -- separate ticket is OK by me.

Reply:
Cool. I'll remove this comment in favour of https://github.com/elastic/apm-agent-nodejs/issues/2151